src/common/virtual_file_system.cpp

```diff
@@ -216,6 +219,40 @@ bool VirtualFileSystem::SubSystemIsDisabled(const string &name) {
 	return disabled_file_systems.find(name) != disabled_file_systems.end();
 }
 
+FileSystem &VirtualFileSystem::FindFileSystem(const string &path, optional_ptr<FileOpener> opener) {
+	return FindFileSystem(path, FileOpener::TryGetClientContext(opener));
+}
+
+FileSystem &VirtualFileSystem::FindFileSystem(const string &path, optional_ptr<ClientContext> client_context) {
+	auto &fs = FindFileSystemInternal(path, !!client_context);
+
+	if (!disabled_file_systems.empty() && disabled_file_systems.find(fs.GetName()) != disabled_file_systems.end()) {
+		throw PermissionException("File system %s has been disabled by configuration", fs.GetName());
+	}
+
+	if (client_context && fs.IsAutoloadingFileSystem()) {
```

Comment:
Maybe we look for the FS name instead?

Reply:
Thanks! I did change from this being a sub-FileSystem with very limited capabilities to handle this as part of VirtualFileSystem capabilities, so this comment is not outdated, but pointed in the same direction I think.